### ISSUE TITLE
rustdoc: Show span of link definition for `broken_intra_doc_links`

### ISSUE THREAD

**Issue by camelid** [rust-lang/rust#82210]
### Code

```rust
//! Look at [foo].
//!
//! [foo]: Nonexistent
```

### Error

```
warning: unresolved link to `Nonexistent`
 --> foo.rs:1:13
  |
1 | //! Look at [foo].
  |             ^^^^^ no item named `Nonexistent` in scope
  |
  = note: `#[warn(broken_intra_doc_links)]` on by default
  = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`
```

Instead (or in addition) it should point to `[foo]: Nonexistent`. The way it is now is confusing because you can't see `Nonexistent` anywhere in the span that is pointed to.

Not sure if an issue for this already exists, so I filed this one.

**Comment by jyn514** [comment]
Oh ugh, https://github.com/rust-lang/rust/pull/80550 never got backported. This is fixed on beta.

**Comment by jyn514** [comment]
Personally I don't think this is worth a stable-backport.

**Comment by jyn514** [comment]
This is fixed in 1.51.0.
